### PR TITLE
Improve console output of compile task

### PR DESCRIPTION
**Pull request by fvictorio**
image (2)

@fzeoli what do you think?

### CODE REVIEW

**Comment by fvictorio** [comment]
There is a red mark shown when a compilation fails, but not when downloading fails. Maybe we should add that too.

**Comment by fzeoli** [comment]
fewer lines would be nice but this is good enough for now given circumstance

**Comment by alcuadrado** [comment]
This shows both a check and a cross: https://github.com/nomiclabs/buidler/pull/819/checks?check_run_id=1173879518#step:8:105

**Comment by fvictorio** [comment]
> This shows both a check and a cross

Fixed. There was also another problem: a cross was being shown if there were compilation warnings. That works fine now.

File: packages/hardhat-core/src/builtin-tasks/compile.ts [hunk 1]
```diff
@@ -394,25 +410,94 @@ export default function () {
...
+      }: {
+        isCompilerDownloaded: boolean;
+        quiet: boolean;
+        solcVersion: string;
+      }) => {
+        if (quiet || isCompilerDownloaded) {
+          return;
+        }
+
+        process.stdout.write(`Downloading compiler ${solcVersion}... `);
```
Comment (alcuadrado) [inline packages/hardhat-core/src/builtin-tasks/compile.ts:431]:
```suggestion
        process.stdout.write(`Downloading solcjs ${solcVersion}... `);
```

**Pull request merged**